compiler/qsc_rir/src/rir.rs

```diff
@@ -238,14 +251,45 @@ impl Display for ConditionCode {
     }
 }
 
-impl Display for CallableType {
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum FcmpConditionCode {
+    False,
+    OrderedAndEqual,
+    OrderedAndGreaterThan,
+    OrderedAndGreaterThanOrEqual,
+    OrderedAndLessThan,
+    OrderedAndLessThanOrEqual,
+    OrderedAndNotEqual,
+    Ordered,
+    UnorderedOrEqual,
+    UnorderedOrGreaterThan,
+    UnorderedOrGreaterThanOrEqual,
+    UnorderedOrLessThan,
+    UnorderedOrLessThanOrEqual,
+    UnorderedOrNotEqual,
+    Unordered,
+    True,
+}
```

Comment:
nitpick: looks like we only ever use the ordered variants here, are the others just included for completeness? Just wondering if it's worth scoping down to just the ones we produce from codegen. Not really important, there's no harm in having them here, mostly curious as to the motivation.

Reply:
Completeness is the reason. It didn't feel right to map half of an LLVM enum.